ts/features/wallet/utils/index.tsx

```diff
@@ -39,3 +48,24 @@ export const renderWalletCardFn = (
     />
   ) : null;
 };
+
+/**
+ * A higher-order component which renders a component only if the category filter matches the given category
+ * @param category - The category to filter by
+ * @param WrappedComponent - The component to render
+ * @returns The component or null if the category filter does not match
+ */
+export const withWalletCategoryFilter =
+  <P,>(
```

Comment:
```suggestion
 <P extends Record<string, unknown>>(
```

If we constrain `P` to be an object we can avoid casting to `any` in `{...(props as any)}`.

Reply:
@gispada absolutely, thanks for the suggestion! Addressed here https://github.com/pagopa/io-app/pull/6570/commits/6703d288d1b32133404d56b98f2a4d9d48346ad3